src/server/game/Entities/Player/Player.cpp

```diff
@@ -18756,9 +18759,26 @@ bool Player::LoadFromDB(ObjectGuid guid, CharacterDatabaseQueryHolder* holder)
 
     m_achievementMgr->CheckAllAchievementCriteria(this);
     m_questObjectiveCriteriaMgr->CheckAllQuestObjectiveCriteria(this);
+
+    PushQuests();
     return true;
 }
 
+void Player::PushQuests()
+{
+    std::vector<Quest const*> const& questsAutoPush = sObjectMgr->GetQuestTemplatesAutoPush();
+
+    for (Quest const *quest : questsAutoPush)
+    {
+        //
+        if (quest->GetQuestTag() && quest->GetQuestTag() != QuestTagType::Tag)
+            continue;
+
+        if (!quest->IsUnavailable() && CanTakeQuest(quest, false))
+            AddQuest(quest, nullptr);
```

Comment:
Shouldn't this be `AddQuestAndCheckCompletion(quest, nullptr);`?

Reply:
Usually I would say no, but I noticed the SL quest progress is kept (both tc and on retail), so yes. Either way it's probably safer to use it than not.